File: java/strategy/DuckSimulator/client/RedheadDuck.java
package client;

import behavior.fly.FlyWithWings;
import behavior.quack.Squeak;

public class RedheadDuck extends Duck {

    public RedheadDuck(String name) {
        super(name);
        flyBehavior = new FlyWithWings(name);
        quackBehavior = new Squeak(name);
    }
}

File: kotlin/strategy/DuckSimulator/client/RedheadDuck.kt
package client

import behavior.fly.FlyWithWings
import behavior.quack.Squeak

class RedheadDuck(duckName: String) : Duck(duckName) {

    init {
        flyBehavior = FlyWithWings(duckName)
        quackBehavior = Squeak(duckName)
    }

}